### PR TITLE
Supported token for remote CSV import

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Supported token for remote CSV import

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -1802,12 +1802,23 @@ read_delim_file <- function(file, delim, quote = '"',
...
   loadNamespace("readr")
   loadNamespace("stringr")
   if (stringr::str_detect(file, "^https://") ||
       stringr::str_detect(file, "^http://") ||
       stringr::str_detect(file, "^ftp://")) {
+    if(!is.null(token_key)) {
+      token <- exploratory::getTokenInfo(token_key)
+      if(!is.null(token)) {
+        if(stringr::str_detect(file, "\\?") || stringr::str_detect(file, "\\&")) {
+          file <- stringr::str_c(file, "&", token_key, "=", token)
```
Comment (hidekoji) [inline R/system.R]: change token_key to some predefined name.

File: R/system.R [hunk 2]
```diff
@@ -1802,12 +1802,23 @@ read_delim_file <- function(file, delim, quote = '"',
                             locale = readr::default_locale(),
                             na = c("", "NA"), quoted_na = TRUE,
                             comment = "", trim_ws = FALSE,
-                            skip = 0, n_max = Inf, guess_max = min(1000, n_max), progress = interactive()){
+                            skip = 0, n_max = Inf, guess_max = min(1000, n_max), token_key = NULL,
```
Comment (hidekoji) [inline R/system.R]: Remove `token_key` argument.

File: R/system.R [hunk 3]
```diff
@@ -1802,12 +1802,24 @@ read_delim_file <- function(file, delim, quote = '"',
...
                             comment = "", trim_ws = FALSE,
-                            skip = 0, n_max = Inf, guess_max = min(1000, n_max), progress = interactive()){
+                            skip = 0, n_max = Inf, guess_max = min(1000, n_max),
+                            progress = interactive(), with_api_key = FALSE){
   loadNamespace("readr")
   loadNamespace("stringr")
   if (stringr::str_detect(file, "^https://") ||
       stringr::str_detect(file, "^http://") ||
       stringr::str_detect(file, "^ftp://")) {
+    if(with_access_token){
```
Comment (hideaki) [inline R/system.R]: It should be with_api_key.

Comment (hidekoji) [reply R/system.R]: Updated variable name.

**Pull request merged**